components/base/sensorcontrolled/spin.go

```diff
@@ -105,162 +52,88 @@ func (sb *sensorBase) stopSpinWithSensor(
 		timeOut = 10 * time.Second
 	}
 
-	sb.activeBackgroundWorkers.Add(1)
-	utils.ManagedGo(func() {
-		ticker := time.NewTicker(yawPollTime)
-		defer ticker.Stop()
+	for {
+		// check if we want to poll the sensor at all
+		// other API calls set this to false so that this for loop stops
 
-		for {
-			// check if we want to poll the sensor at all
-			// other API calls set this to false so that this for loop stops
-			if !sb.isPolling() {
-				ticker.Stop()
-			}
-
-			if err := ctx.Err(); err != nil {
-				ticker.Stop()
-				return
-			}
+		if !sb.isPolling() {
```

Comment:
ah I see maybe here it's necessary I'm still not sure tho if we need this interrupted by another api thing. shouldn't spin be blocking til it's done so not interruptible?

Reply:
I was keeping the behavior to have spin be interrupted by other API calls on the base, but I would be very happy to remove all of the polling logic too.

Reply:
it seems to me that spin shouldn't be interruptible so I agree, but rand might have more insight on why it should stay